### ISSUE TITLE
ExceptionCallback fails with Clang AddressSanitizer (in Xcode at least)

### ISSUE THREAD

**Issue by snej** [capnproto/capnproto#1427]
It appears that the ExceptionCallback constructor's check for whether it's on the stack does not work correctly when running under the Clang AddressSanitizer. When I tried to use ExceptionCallback by making a simple subclass and putting an instance in a local variable, the constructor failed with `ExceptionCallback must be allocated on the stack`, even though it _is_ on the stack.

After some experimentation I found that the AddressSanitizer's "detect use of stack after return" option seems to introduce big gaps (around 4MB?) in between stack frames. This breaks the check in the constructor.

For instance, this little test:
```c++
class Foo {
public:
    Foo() {
        char stackVar;
        ptrdiff_t offset = reinterpret_cast<char*>(this) - &stackVar;
        std::cerr << "offset = " << offset << std::endl;
    }
};
int main() {
    Foo foo;
}
```
logs `offset = 4068352` when run with Xcode's "detect use of stack after return" checkbox enabled, but `offset = 352` without it.

### Steps To Reproduce
```c++
#include <kj/exception.h>
int main() {
    ExceptionCallback x;
}
```
Compile with Clang, with AddressSanitizer enabled, and the "detect use of stack after return" enabled (I'm not sure what the command-line flag for that is.)

### Workaround

Turn off "detect use of stack after return". Unfortunately this is a really useful check that has saved my bacon many a time.

Suggestion: It would be nice if the assertion in ExceptionCallback could be disabled at runtime somehow. I'm using Cap'n Proto installed by my package manager so it's not convenient for me to recompile it after commenting out that line...

### Configuration
MacBook Pro (ARM M1, 2021)
Xcode 13.3b2

**Comment by kentonv** [comment]
Strangely, we haven't run into this problem in our own ASAN builds, but I haven't checked if we're using that flag.

It looks like there's an existing `#ifdef` to skip the check in fuzzing builds. I think it would make sense to do the same if ASAN is enabled, which I believe you can query via built-in preprocessor defines. Happy to accept a PR.

**Comment by kentonv** [comment]
(Sorry, didn't mean to close, hit the wrong button.)

**Comment by andronat** [comment]
I'm having the same issue. Any plans for a fix?

**Comment by kentonv** [comment]
As a matter of fact, this was fixed in #1478 -- once we started hitting the issue on Cloudflare Workers (after a toolchain update I guess), I was able to address it.

I suppose I should do a 0.10.2 patch release with this and #1477.

**Comment by kentonv** [comment]
I've released 0.10.2 with the fix for this. Download links on the site: https://capnproto.org/install.html
